specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,316 @@
+                    return S_OK;
+                })
+                .Get(),
+            &m_launchingExternalProtocolToken));
+    }
+}
+    
+```
+
+## .NET and WinRT
+
+```c#
+private WebView2 webView;
+void RegisterLaunchingExternalProtocolHandler() 
+{
+    webView.CoreWebView2.LaunchingExternalProtocol += (sender, args) {
+        {
+            CoreWebView2Deferral deferral = args.GetDeferral();
+            System.Threading.SynchronizationContext.Current.Post((_) =>
+            {
+                using (deferral)
+                {
+                    if (String.Equals(args.Uri, "calculator:///", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // If this matches our desired protocol, then set the
+                        // event args to cancel the event and launch the
+                        // calculator app.
+                        ProcessStartInfo info = new ProcessStartInfo
+                        {
+                            FileName = args.Uri,
+                            UseShellExecute = true
+                        };
+                        Process.Start(info);
+                        args.Cancel = true;
+                    } 
+                    else
+                    {
+                        // Otherwise use a deferral to display a custom dialog
+                        // in which the user can choose to grant permission
+                        // to launch the app and set the event args accordingly. 
+                        string text = "Launching External Protocol";
+                        if (args.InitiatingUri != "")
+                        {
+                            text += "from ";
+                            text += args.InitiatingUri;
+                        }
+                        text += " to ";
+                        text += args.Uri;
+                        text += "\n";
+                        text += "Do you want to grant permission?";
+                        string caption = "Launching External Protocol request";
+                        MessageBoxButton btnMessageBox = MessageBoxButton.YesNoCancel;
+                        MessageBoxImage icnMessageBox = MessageBoxImage.None;
+                        MessageBoxResult resultbox = MessageBox.Show(text, caption, btnMessageBox, icnMessageBox);
+                        switch (resultbox)
+                        {
+                            case MessageBoxResult.Yes:
+                                ProcessStartInfo info = new ProcessStartInfo
+                                {
+                                    FileName = args.Uri,
+                                    UseShellExecute = true
+                                };
+                                Process.Start(info);
+                                args.Cancel = true;
+                                break;
+
+                            case MessageBoxResult.No:
+                                args.Cancel = true;

```

Comment:
Is this sample setting args.Cancel = true for all the cases? Every case in the switch but also in the if and the else blocks? Can you just set args.Cancel at the beginning with a comment explaining that we set cancel in order to either silently launch the trusted calculator app or to show our own custom dialog?